optaplanner-core/src/main/java/org/optaplanner/core/config/heuristic/selector/entity/pillar/PillarSelectorConfig.java

```diff
@@ -82,41 +95,93 @@ public void setMaximumSubPillarSize(Integer maximumSubPillarSize) {
 
     /**
      * @param configPolicy never null
+     * @param pillarType if null, defaults to {@link PillarType#FULL_AND_SUB} for backwards compatibility reasons.
+     * @param pillarOrderComparatorClass if not null, will force entites in the pillar to come in this order
      * @param minimumCacheType never null, If caching is used (different from {@link SelectionCacheType#JUST_IN_TIME}),
      * then it should be at least this {@link SelectionCacheType} because an ancestor already uses such caching
      * and less would be pointless.
      * @param inheritedSelectionOrder never null
      * @param variableNameIncludeList sometimes null
      * @return never null
      */
-    public PillarSelector buildPillarSelector(HeuristicConfigPolicy configPolicy,
-            SelectionCacheType minimumCacheType, SelectionOrder inheritedSelectionOrder,
-            List<String> variableNameIncludeList) {
+    public PillarSelector buildPillarSelector(HeuristicConfigPolicy configPolicy, PillarType pillarType,
+            Class<? extends Comparator> pillarOrderComparatorClass, SelectionCacheType minimumCacheType,
+            SelectionOrder inheritedSelectionOrder, List<String> variableNameIncludeList) {
+        if (subPillarEnabled != null) {
+            if (pillarType == null) {
+                LOGGER.warn("Property subPillarEnabled ({}) on pillarSelectorConfig ({}) is deprecated for removal.\n" +
+                        "Use property pillarType on the parent MoveSelectorConfig.", subPillarEnabled, this);
```

Comment:
The compiler logs a warning when using deprecated methods.
Developers hopefully notice it there - or when following the upgrade recipe.
The developers who don't notice the compilation message, nor look at the upgrade recipe, will feel it when migrating to 8.0 and fix it then.

Generally, optaplanner-core doesn't add a warn log message for deprecated methods at runtime (especially if the deprecated method still behave equally well), because it could cause sys ops to reduce logging verbosity to ignore it (and that means they go blind on what optaplanner is doing, which is worse).